src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp

```diff
@@ -3624,6 +3625,7 @@ void MusicXMLParserPass2::barline(const QString& partId, Measure* measure, const
     while (_e.readNextStartElement()) {
         if (_e.name() == "bar-style") {
             barStyle = _e.readElementText();
+            barlineColor = _e.attributes().value("color").toString();
```

Comment:
I believe this is the wrong order, causing color import not to work
At least with that it works for double and end-barlines